src/fill.c

```diff
@@ -246,19 +246,23 @@ void stop_bad_direction() {
 }
 
 static
-int parse_max_fill(SEXP x) {
+int parse_max_fill(r_obj* x) {
   if (x == R_NilValue) {
     return INFINITE_FILL;
   }
 
-  x = PROTECT(vec_cast(x, vctrs_shared_empty_int, args_max_fill, args_empty));
+  x = KEEP(vec_cast(x,
+                    vctrs_shared_empty_int,
+                    args_max_fill,
+                    args_empty,
+                    r_lazy_null));
```

Comment:
So eventually we will go back through places like this and pass the call through?

Reply:
yeah, that's a lot of repetitive work and perhaps not worth it in some cases. In the same way that in some places we haven't passed an `arg` either.